### ISSUE TITLE
训练准确率一直为0.。。。

### ISSUE THREAD

**Issue by cendelian** [WenmuZhou/PytorchOCR#120]
大佬，我按照文档配置自己的路径之类，
训练准确率一直为0，，，，
eval阶段准确率也是0.。。。。
![image](https://user-images.githubusercontent.com/50360389/99934507-25a8d580-2d99-11eb-9efa-d1f8eda29550.png)
![image](https://user-images.githubusercontent.com/50360389/99934509-25a8d580-2d99-11eb-8d23-4409499532dc.png)
![image](https://user-images.githubusercontent.com/50360389/99934535-38230f00-2d99-11eb-893d-2edde17d63f6.png)

**Comment by cendelian** [comment]
使用的是绝对路径。图片读取，label读取是正确的。。。，

**Comment by novioleo** [comment]
麻烦查看下你的卷积层中哪个环节数据错了。你这个大概率就是图片加载错误，或者其他类型的错误。简而言之输入到网络中的图大概率有nan

**Comment by cendelian** [comment]
> 麻烦查看下你的卷积层中哪个环节数据错了。你这个大概率就是图片加载错误，或者其他类型的错误。简而言之输入到网络中的图大概率有nan

数据集加载没有问题，此外，换了数据集，换成IIIT5K数据集，训练准确率也是一直都是000000000

**Comment by novioleo** [comment]
那你能debug看下是哪个环节不正常的吗？你这样说我们没法判断。 @cendelian

**Comment by WenmuZhou** [comment]
建议先使用数据数据集看看，IIIT5K数据集太小，欠拟合也正常

**Comment by cendelian** [comment]
> 建议先使用数据数据集看看，IIIT5K数据集太小，欠拟合也正常

我打印输出如下：
```
class RecMetric:
    def __init__(self, converter):
        """
        文本识别相关指标计算类

        :param converter: 用于label转换的转换器
        """
        self.converter = converter

    def __call__(self, predictions, labels):
        n_correct = 0
        norm_edit_dis = 0.0
        predictions = predictions.softmax(dim=2).detach().cpu().numpy()
        # print('prediction is {}'.format(predictions))
        preds_str = self.converter.decode(predictions)
        print('preds_str is {}'.format(preds_str))
```

训练一开始 pred_str有数据，过了两三次就输出都是空
![image](https://user-images.githubusercontent.com/50360389/100223489-85001480-2f56-11eb-8674-e132160790dd.png)
。




我的digit.txt 一共有62个字符（IIIT5K 62个字符）如下：
![image](https://user-images.githubusercontent.com/50360389/100223722-d4464500-2f56-11eb-8718-29896221ec3a.png)

**Comment by cendelian** [comment]
> 那你能debug看下是哪个环节不正常的吗？你这样说我们没法判断。 @cendelian

大佬，rec_train.py是不是有bug？。。
梯度清零可以在模型推理前吗？

我把                # 清零梯度及反向传播
                optimizer.zero_grad()
移至
                loss_dict['loss'].backward()
                torch.nn.utils.clip_grad_norm_(net.parameters(), 5)
前
训练就不是准确率一直为0了，，，，
![image](https://user-images.githubusercontent.com/50360389/100228416-9b5d9e80-2f5d-11eb-93ee-a5bc61b0f08a.png)

**Comment by Aionrichman** [comment]
> > 那你能debug看下是哪个环节不正常的吗？你这样说我们没法判断。 @cendelian
> 
> 大佬，rec_train.py是不是有bug？。。
> 梯度清零可以在模型推理前吗？
> 
> 我把 # 清零梯度及反向传播
> optimizer.zero_grad()
> 移至
> loss_dict['loss'].backward()
> torch.nn.utils.clip_grad_norm_(net.parameters(), 5)
> 前
> 训练就不是准确率一直为0了，，，，
> ![image](https://user-images.githubusercontent.com/50360389/100228416-9b5d9e80-2f5d-11eb-93ee-a5bc61b0f08a.png)

请问移动后代码是怎样的？可以说详细点吗

**Comment by cendelian** [comment]
> > > 那你能debug看下是哪个环节不正常的吗？你这样说我们没法判断。 @cendelian
> > 
> > 
> > 大佬，rec_train.py是不是有bug？。。
> > 梯度清零可以在模型推理前吗？
> > 我把 # 清零梯度及反向传播
> > optimizer.zero_grad()
> > 移至
> > loss_dict['loss'].backward()
> > torch.nn.utils.clip_grad_norm_(net.parameters(), 5)
> > 前
> > 训练就不是准确率一直为0了，，，，
> > ![image](https://user-images.githubusercontent.com/50360389/100228416-9b5d9e80-2f5d-11eb-93ee-a5bc61b0f08a.png)
> 
> 请问移动后代码是怎样的？可以说详细点吗
我发现跟这个没有关系。。。。。训练几轮准确率就有变化，，，，准确率很高，但是评估的时候很不准。。。。。。。。。
